src/Warble/warble_service.cc

```diff
@@ -299,4 +316,59 @@ PayloadOptional WarbleService::ReadThread(const Payload &payload,
   reply_payload.PackFrom(reply);
   return PayloadOptional(reply_payload);
 }
+
+PayloadOptional WarbleService::Stream(const Payload &payload, 
+                                       const StoragePtr &kv_store) {
+  StreamRequest request;
+  StreamReply reply;
+  Payload reply_payload;
+  StringVector key_vector;
+  payload.UnpackTo(&request);
+  std::string hashtag_key = kHashtagPrefix + request.hashtag();;
+  Timestamp time = request.time();
+  int startTime = time.seconds();
+  
+  StringVector k = {hashtag_key};
+  std::string string_ids = kv_store->Get(k)[0].value_or("");
+  if (string_ids != "") {
+    StringVector vector_ids = deserialize(string_ids, ',');
+    for (auto s : vector_ids) {
+      key_vector.push_back(kWarblePrefix + s);
+    }
+    StringOptionalVector warbles_opt_vector = kv_store->Get(key_vector);
+    LOG(INFO) << "The number of warbles containing hastag:" << warbles_opt_vector.size()<<std::endl;
+    if (warbles_opt_vector.size() > 0) {
+      for (auto op : warbles_opt_vector) {
+        Warble temp;
+        temp.ParseFromString(op.value());
+        int w_time = temp.timestamp().seconds();
+        if (w_time > startTime) {
+          auto w = reply.add_warbles();
```

Comment:
Use type deduction only if it makes the code clearer to readers who aren't familiar with the project, or if it makes the code safer. (https://google.github.io/styleguide/cppguide.html#Type_deduction)
Therefore, I think it should be better to use the explicit type if you know it.